traffic_ops/testing/api/assert/assertions.go

```diff
@@ -16,14 +16,36 @@ package assert
 */
 
 import (
+	"fmt"
 	"reflect"
 	"testing"
 )
 
+// failureOutput checks if there is a message to be parsed and concatenates with a failure message.
+func failureOutput(failureMessage string, msgAndArgs ...interface{}) string {
+	output := failureMessage
+	message := ""
+	if len(msgAndArgs) == 1 {
+		msg := msgAndArgs[0]
+		if msgAsStr, ok := msg.(string); ok {
+			message = msgAsStr
+		}
+		message = fmt.Sprintf("%+v", msg)
+	}
+	if len(msgAndArgs) > 1 {
+		message = fmt.Sprintf(msgAndArgs[0].(string), msgAndArgs[1:]...)
+	}
+	if len(message) > 0 {
+		output = "\n" + output + "\nMessages: " + message + "\n"
```

Comment:
I'm against putting newline characters in most kinds of output, including test failures. Line-based output is extremely common and easy to parse, so most tools work well with it, instead of with custom multi-line formats.